core/src/test/java/com/linecorp/armeria/client/AdditionalAuthorityTest.java

```diff
@@ -90,72 +98,70 @@ void additionalAuthorityHasHighestPrecedence() {
+                                                                 .authority("bar:" + serverAPort)
+                                                                 .build());
+        assertThat(client.execute(request).contentUtf8())
+                .isEqualTo("bar/bar:" + serverAPort);
     }
 
     @Test
-    void shouldIgnoreInvalidAdditionalAuthority() {
-        // Missing a closing bracket
-        try (SafeCloseable ignored = Clients.withContextCustomizer(
-                ctx -> ctx.addAdditionalRequestHeader(HttpHeaderNames.AUTHORITY, "[::1"))) {
-
-            assertThatThrownBy(() -> client.get("http://foo:" + serverAPort))
-                    .isInstanceOf(UnprocessedRequestException.class)
-                    .hasCauseInstanceOf(IllegalArgumentException.class)
-                    .hasMessageContaining("Invalid bracketed host/port");
-        }
-
-        // Port only
-        try (SafeCloseable ignored = Clients.withContextCustomizer(
-                ctx -> ctx.addAdditionalRequestHeader(HttpHeaderNames.AUTHORITY, ":8080"))) {
-
-            final HttpRequest request = HttpRequest.of(RequestHeaders.builder(HttpMethod.GET, "/")
-                                                                     .scheme("http")
-                                                                     .authority("bar:" + serverAPort)
-                                                                     .build());
-
-            assertThatThrownBy(() -> client.execute(request))
-                    .isInstanceOf(UnprocessedRequestException.class)
-                    .hasCauseInstanceOf(IllegalArgumentException.class)
-                    .hasMessageContaining("Not a valid domain name");
-        }
-
-        // Missing a port number
-        try (SafeCloseable ignored = Clients.withContextCustomizer(
-                ctx -> ctx.addAdditionalRequestHeader(HttpHeaderNames.AUTHORITY, "foo:"))) {
-
-            final HttpRequest request = HttpRequest.of(RequestHeaders.builder(HttpMethod.GET, "/")
-                                                                     .scheme("http")
-                                                                     .authority("bar:" + serverAPort)
-                                                                     .build());
-
-            assertThatThrownBy(() -> client.execute(request))
-                    .isInstanceOf(UnprocessedRequestException.class)
-                    .hasCauseInstanceOf(IllegalArgumentException.class)
-                    .hasMessageContaining("Missing port number");
-        }
+    void shouldIgnoreAuthorityInRequestWithBaseUriClient() {
+        final HttpRequest request = HttpRequest.of(RequestHeaders.builder(HttpMethod.GET, "/")
+                                                                 .scheme("http")
+                                                                 .authority("bar:" + serverAPort)
```

Comment:
Question: I think it's slightly confusing that authority is overridden depending on where the header is defined. Having said this, I don't really have a good idea on how to solve this 😅   

(I guess the most realistic way is to store `authority` not in headers but a separate field in `ClientOptions`)

```
...
HttpRequest.of(RequestHeaders.builder(HttpMethod.GET, "/")
                             .authority("bar:" + serverAPort) // doesn't override
...
WebClient.builder("http://foo:" + serverAPort)
         .setHeader(HttpHeaderNames.AUTHORITY, "bar:" + serverAPort) // does override
         .authority("bar: " + serverAPort) // does override
...
```

I don't think this has to be done in this PR, but I just wanted to mention it.

Reply:
It is a bit tricky. We can't use `:authoriy` header directly due to some limitations. 
Let's say a simple `HttpService` proxies an incoming request to another server using Armeria's `WebClient`.
We may implement the code like:
```java
class MyProxyService implements HttpService {
    
    private WebClient nonBaseUriClient = WebClient.of();
    private WebClient baseUriClient = WebClient.of("https://backend.com");
    public HttpResponse serve(ServiceRequestContext ctx, HttpRequest req) {
        // 1. Proxies the request using nonBaseUriClient which creates
        // an `Endpoint` using the `:authority` header. 
        return nonBaseUriClient.execute(req);
        // 2. Proxies the request using nonBaseUriClient which creates
        // an `Endpoint` using the base URI header and the authority header is discarded. 
        return baseUriClient.execute(req);
    }
}
```

`:authority` could get overridden if a base URI is set to `WebClient`.
We set these rules because some users want requests to follow the authority and others need to redirect the request to a specific destination. 

If a custom `:authority` is set to `HttpHeaders`, we can't determine whether it could be overwritten by the base URI or not.
As of this PR, an additional `:authority` is always respected regardless of the existence of the base URI of `WebClient`.

Reply:
I see, to be clear I wasn't really proposing that we allow overriding the authority from `HttpHeaders`.
Rather, I was proposing the following:

```
...
HttpRequest.of(RequestHeaders.builder(HttpMethod.GET, "/")
                             .authority("bar:" + serverAPort) // doesn't override
...
WebClient.builder("http://foo:" + serverAPort)
         .setHeader(HttpHeaderNames.AUTHORITY, "bar:" + serverAPort) // doesn't override
         .authority("bar: " + serverAPort) // does override
...
```

So setting by headers don't override the authority consistently from a user's point of view.

Reply:
Alternatively, the naming could also be different to further reduce confusion.

```
...
HttpRequest.of(RequestHeaders.builder(HttpMethod.GET, "/")
                             .authority("bar:" + serverAPort) // doesn't override
...
WebClient.builder("http://foo:" + serverAPort)
         .setHeader(HttpHeaderNames.AUTHORITY, "bar:" + serverAPort) // doesn't override
         .overrideAuthority("bar: " + serverAPort) // does override
...
WebClient.prepare().overrideAuthority() // does override
```

Again, I don't think this has to be done in this PR but this is just a suggestion 😅

Reply:
Renaming to `overrideAuthority` is a nice idea. 👍

Reply:
It is a good suggestion. The API naming and signature do not explain the behavior.
It is clear from our API that additional headers have higher priority and they always override the old headers before being sent.

I'm thinking of adding a `boolean` which is whether to override the original `authority()` header.
For example:
```
String customAuthority = ...;
WebClient.builder()
          // does override because they are additional headers
         .setHeader(HttpHeaderNames.AUTHORITY, customAuthority)
         .authority(customAuthority, true) // does override 
         .authority(customAuthority, false) // does not override if exists but use when empty.
```

Reply:
I rescheduled this PR to 1.21.0 to develop more ideas. 😆